lighthouse-treemap/app/src/main.js

```diff
@@ -64,10 +64,20 @@ class TreemapViewer {
     /** @type {WeakMap<LH.Treemap.Node, LH.Treemap.NodePath>} */
     this.nodeToPathMap = new WeakMap();
 
-    this.documentUrl = options.lhr.requestedUrl;
+    this.documentUrl = new URL(options.lhr.requestedUrl);
     this.el = el;
     this.getHueForD1NodeName = TreemapUtil.stableHasher(TreemapUtil.COLOR_HUES);
 
+    // These depth one node uses the network URL for the name, but we want
+    // to elide common parts of the URL so text fits better in the UI.
+    for (const node of this.depthOneNodesByGroup.scripts) {
+      const url = new URL(node.name);
+      node.name = TreemapUtil.elideUrl(url, this.documentUrl);
+      if (url.href === this.documentUrl.href) {
```

Comment:
This works as temporary solution, but misses the case of iframes's inline JS. Haven't ever tested something like that so no idea if the app even works :)